File: sdk-node/examples/properties/retrieve.ts
/**
 * Retrieve a single property by ID. `Select One` and `Select Multiple`
 * properties additionally carry an `options` array.
 *
 * Run:
 *   npx tsx examples/properties/retrieve.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const property = await client.properties.retrieve('prop_replace_with_real_id')

console.log(`property ${property.id} [${property.type}]`)
console.log(`  name     ${property.name}`)
console.log(`  object   ${property.objectType}`)
console.log(`  status   ${property.status}`)
if (property.type === 'Select One' || property.type === 'Select Multiple') {
  for (const option of property.options) {
    console.log(`  option   ${option.label} -> ${option.value}`)
  }
}

File: sdk-python/examples/properties/retrieve.py
"""Run with: python examples/properties/retrieve.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        prop = client.properties.retrieve("prop_replace_with_real_id")
        print(f"property {prop.id} [{prop.type}]")
        print(f"  name        {prop.name}")
        print(f"  object_type {prop.object_type}")
        print(f"  status      {prop.status}")
        for option in prop.options or []:
            print(f"  option      {option.value} -> {option.label}")


if __name__ == "__main__":
    main()
